dist/src/test/java/io/camunda/zeebe/shared/management/ControlledActorClockEndpointTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.shared.management;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.byLessThan;
+
+import io.camunda.zeebe.shared.management.ActorClockEndpoint.Response;
+import io.camunda.zeebe.util.sched.clock.ControlledActorClock;
+import java.time.Instant;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.InstanceOfAssertFactory;
+import org.assertj.core.api.ObjectAssert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+final class ControlledActorClockEndpointTest {
+
+  private final InstanceOfAssertFactory<Response, ObjectAssert<Response>> instanceOfRecord =
+      new InstanceOfAssertFactory<>(Response.class, Assertions::assertThat);
+
+  private final ActorClockEndpoint endpoint =
+      new ActorClockEndpoint(new ControlledActorClockService(new ControlledActorClock()));
+
+  @BeforeEach
+  private void resetClock() {
+    endpoint.resetTime();
+  }
+
+  @Test
+  void canRetrieveCurrentClock() {
+    final var response = endpoint.getCurrentClock();
+    assertThat(response.getStatus()).isEqualTo(200);
+  }
+
+  @Test
+  void canPinMutableClock() {
+    // given
+    final var millis = 1635672964533L;
+
+    // when
+    final var response = endpoint.modify("pin", millis, null);
+
+    // then
+
+    assertThat(response.getStatus()).isEqualTo(200);
+    assertThat(response.getBody())
+        .asInstanceOf(instanceOfRecord)
+        .satisfies((body) -> assertThat(body.epochMilli).isEqualTo(millis))
+        .isNotNull();
+  }
+
+  @Test
+  void canOffsetMutableClock() {
+    // given
+    final var offset = 10000L;
+
+    // when
+    final var response = endpoint.modify("add", null, offset);
+
+    // then
+    assertThat(response.getStatus()).isEqualTo(200);
+    assertThat(response.getBody())
+        .isNotNull()
+        .asInstanceOf(instanceOfRecord)
+        .satisfies(
+            (body) ->
+                assertThat(body.epochMilli)
```

Comment:
:upside_down_face: Using the `Instant` variant (e.g. `assertThat(body.instant).isCloseTo(Instant.now(), byLessThan(offset + 1, ChronoUnit.MILLIS))`) gives a nicer error message in my opinion (e.g. expected TIME A to be within 10 millis of TIME B), but both are equally readable, so I'll leave it up to you.